front/src/applications/operationalStudies/components/SimulationResults/SpeedSpaceChart/SpeedSpaceChart.tsx

```diff
@@ -57,27 +71,50 @@ export default function SpeedSpaceChart(props) {
   const [rotate, setRotate] = useState(false);
   const [showSettings, setShowSettings] = useState(false);
 
-  const ref = useRef();
+  const ref = useRef<HTMLDivElement>(null);
 
-  const onLocalSetSettings = (settings) => {
+  const dispatch = useDispatch();
+
+  const dispatchUpdateMustRedraw = (newMustRedraw: boolean) => {
+    dispatch(updateMustRedraw(newMustRedraw));
+  };
+
+  const dispatchUpdateTimePositionValues = (newTimePositionValues: string) => {
+    dispatch(updateTimePositionValues(newTimePositionValues));
+  };
+
+  const toggleSetting = (settings: SpeedSpaceSettingsType) => {
+    dispatch(updateSpeedSpaceSettings(settings));
+    dispatch(updateMustRedraw(true));
+  };
+
+  const onLocalSetSettings = (settings: SpeedSpaceSettingsType) => {
     setLocalSettings(settings);
-    onSetSettings(settings);
+    toggleSetting(settings);
   };
 
+  const trainSimulation = useMemo(
+    () =>
+      selectedTrain
+        ? prepareData(selectedTrain)
+        : prepareData(ORSD_GRAPH_SAMPLE_DATA.simulation.present.trains[0]),
+    [selectedTrain]
+  );
```

Comment:
This can be simplified if we know that selectedTrain is not undefined
```suggestion
  const trainSimulation = useMemo(
    () =>
      selectedTrain
        ? prepareData(selectedTrain)
        : prepareData(ORSD_GRAPH_SAMPLE_DATA.simulation.present.trains[0]),
    [selectedTrain]
  );
```

Reply:
Ok so I simplified it here but I had to change getSelectedTrain selector type in selector.ts which was (and still kind is) strange.